### ISSUE TITLE
[Item] ICC Ring Quest Rollback Issue

### ISSUE THREAD

**Issue by TheVisvires** [dalaranwow/dalaran-wow#5484]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behaviour**: Tell us what happens in detail (Where, When, What, Why, Who)
The upgrade quest for the ICC ring isn't atomic. It is possible to have it marked as complete while not upgrading the ring, if the server crashes shortly thereafter.

In ICC, when you turn in Path of Courage to Ormu the Penitent, the quest reward might not persist past an immediate server crash, instead being rolled back, while the quest completion will remain. This prevents progression through the chain because the next level of the quest requires the Revered ring, but I'm still on the Honored ring because the Revered quest bugged. I think this is because I am quite unlucky.

**Expected behaviour**: Tell us what should happen instead
Ideally, the quest gets turned in, the old rep ring disappears from my inventory and the new one is dropped into my inventory.

Turning in the quest should either upgrade the ring and mark the quest as complete, or do neither. If it is going to fail, it would be better to only upgrade the ring without marking the quest as complete. The upgrade of the ring itself would preclude turning the quest in again. 

I suggest having the item swap take place before the quest completion; this allows the possibility of a permanent quest marker instead of a permanent deprivation of a major item. 

**Steps to reproduce the problem**:

1. Achieve reputation with the Ashen Verdict sufficient to upgrade the ring. (Revered)
2. Turn in the upgrade quest.
3. Get spectacularly unlucky on server crash timing, such that the server closes while processing the quest completion and marks the quest as complete while shutting down without granting the ring.
4. Turn in variations of "A Change of Heart" (eg #24839) a couple times to verify that all Path quests of that rep level are similarly locked out.

**Include proofs for this behaviour**
![proof - no chat](https://user-images.githubusercontent.com/39577643/40463332-54fc38da-5eda-11e8-80ce-ff336b0e86c9.jpg)

You can research this on several pages like http://wowhead.com or http://wowwiki.wikia.com or http://www.youtube.com/
Issues without proofs will be closed and tag On Hold will be added until the proof is provided.

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**

Quest is #24834, Path of Courage (The honored-revered one). It's given by NPC #38316, Ormus the Penitent. Ring that it turns in is #50388 "Ashen Band of Greater Courage"; ring that's the reward is #50403 "Ashen Band of Unmatched Courage." Instance is ICC, 10 or 25, #4812. Ashen Verdict is #1156.

You can research this on several pages like https://wowgaming.altervista.org/aowow/ or https://wowhead.com/

**Include Screenshots from the issue if necessary**

Maybe this is one of those things it's easier to fix on the rare times it happens than to prevent. But I've had a ticket open over a month, so I figured I'd drop this in.

Thank you for any assistance.

**Comment by Rushor** [comment]
I will add some some gossips where it will be possible to get back the lost ring.

Moreover please contact a gamemaster which will complete the quest for you, so you can make the progress in your questchain.

![GitHub Logo](https://i.imgur.com/PlMRkvE.jpg)

**Comment by TheVisvires** [comment]
May I ask that status on those gossips? I can't find them. Are they on a different NPC?

**Comment by Chenocide** [comment]
It hasn't been applied to the live realm yet.

**Comment by TheVisvires** [comment]
Thank you.

**Comment by DeckGrek** [comment]
Sorry for necro, but I'm still struggling with a lost ring too. Is there any news so far?
